### PR TITLE
Add YAML support and fix IE support for downloads

### LINKED ISSUE #63: Download is not possible from IE
neither JSON nor Yaml extension is not possible to download.

### PR DESCRIPTION
**Pull request by tulir**
Closes #63

This pull request adds support for downloading the data as a YAML file and should also fix downloading from IE.

@Nazarah Could you test this in IE? There's a demo of this branch available at https://oai.maunium.net/feature/improve-downloads/ If it doesn't work, please send the error information from the browser console.

### CODE REVIEW

**Comment by Nazarah** [comment]
On it

File: src/app.html [hunk 1]
```diff
@@ -8,7 +8,8 @@
     <a href="#/tags">Tags</a>
     <a href="#/paths">Paths</a>
     <a href="#/types">Types</a>
-    <a href="javascript:void(0)" click.delegate="download('json')">Download</a>
+    <a href="javascript:void(0)" click.delegate="download('json')">Download (JSON)</a>
```
Comment (brylie) [inline src/app.html]: Go ahead and nest these strings on their own line. That will improve readability, by letting us scan the text vertically and understand what is going on.

**Comment by Nazarah** [comment]
Tried with MS Edge and compatible versions. (IE9, IE10)
The view didn't load.
Found the following warning in console (snapshot):
image

**Comment by brylie** [comment]
@tulir is this ready for review?

**Comment by tulir** [comment]
IE10+ and Edge should work now.

**Comment by Nazarah** [comment]
Problem with IE9:
Editor doesn't load.
Console shows following error:
image

**Comment by Nazarah** [comment]
Issue with IE10
Editor loads in the page but the console has the following error
image

**Comment by tulir** [comment]
Was there any actual problem in IE10 or just an error in the console?

**Comment by Nazarah** [comment]
@tulir IE10 could load the editor. Could download yaml with typed contents. (added in the previous comment)

**Comment by tulir** [comment]
It seems that IE9 support would require even more polyfills and other such things. Do we really need to support so old browsers? They're just a problem for everyone, including the users who keep using an outdated browser that doesn't get security updates.
I made a pull request that adds a outdated browser notification. See #109

**Comment by brylie** [comment]
> Do we really need to support so old browsers? [IE 8,9, and 10]

I am not an expert, so will defer to other people/data to make a recommendation.

IE < 11 has not been in the top 10 browser versions on the W3Counter's [Browser & Platform Market Share](https://www.w3counter.com/globalstats.php?year=2016&month=7) since July, 2016, at which point it was < 2% adoption:

screenshot_20170526_101124

According to Clicky.com's [Internet explorer (Global marketshare)](https://clicky.com/marketshare/global/web-browsers/internet-explorer/=), IE versions 8, 9, and 10 have flatlined at less than three percent of marketshare:

screenshot_20170526_100220

Larry Maccherone wrote an [Analysis of Browser User/Market-share, or So Long! IE8, 9 and 10](https://blog.pendo.io/2016/01/14/analysis-of-browser-usermarket-share-or-so-long-ie8-9-and-10/
), and made the following relevant recommendation:

> If you are starting a new project now, tell your development teams that they don’t need to support any browser earlier than IE 11.

**Comment by xylix** [comment]
Seems to me like we can drop support for old IE versions. @brylie can you approve this pull request now or inform us if additional changes are needed?

**Pull request merged**